docs/source/migration-two-dot.md

```diff
@@ -219,3 +220,70 @@ server.listen().then(({ url }) => {
   console.log(`🚀 Server ready at ${url}`);
 });
 ```
+
+<h2 id="request-headers">Accessing Request Headers</h2>
+
+Apollo Server 1 allowed request headers to be used in the construction of the GraphQL options. Apollo Server 2 allows constructor to create the context based upon the request.
+
+```js
+//old way
+graphqlExpress((req, res) => ({
+  schema: myGraphQLSchema,
+  context: {
+    token: req.headers['auth-token'],
+  },
+}))
+
+//new way
+new ApolloServer({
+  schema: myGraphQLSchema,
+  context: ({req, res}) => ({
+    token: req.headers['auth-token'],
+  }),
+});
+```
+
+For most other functions that might have required access to the middleware arguments, such as `formatParams`, `formatError`, and `formatResponse`, it is possible to create a `graphql-extension`.
+
+For more complicated use cases, the `ApolloServer` class can be extended to override the `createGraphQLServerOptions` function to create parameters based on the same argument that's passed to the context.
+
+<h2 id="log-function">Replacing `logFunction`</h2>
+
+Apollo Server 2 removes the `logFunction` in favor of using `graphql-extensions`, which provides a more structured and flexible way of instrumenting Apollo Server. The explanation of how to do this more granular logging, can be found in the [metrics section](./features/metrics.html)
+
+<h2 id="graphiql">Replacing GraphiQL</h2>
+
+Apollo Server 2 ships with GraphQL Playground instead of GraphiQL and collocates the gui with the endpoint. GraphQL playground can be customized in the following manner.
```

Comment:
> Apollo Server 2 ships with GraphQL Playground instead of GraphiQL and collocates the gui with the endpoint.

"gui" should be capitalized (it's an acronym), but this sentence could be more clear.  Perhaps:

> Apollo Server 2.0 uses GraphQL Playground (rather than GraphiQL) and, rather than being served on a separate URL, is now served on the same endpoint as the GraphQL API itself (typically, `/graphql`).  This is accomplished without breaking changes for clients by serving the GUI to clients which have requested HTML content using an `Accept` header of `text/html` (the default for user-initiated browser navigation).

...or something.